### PR TITLE
User Guide updates

### PR DESCRIPTION
**Pull request by DellaBitta**
[ Description ]
Updates to the UserGuide.md.  

The guide referred to many stages of development from our older, closed-source version of the library.
Additional changes were made to remove notes about Control Topics, and to add links to our Developer Center.

[ JIRA ]
None.

[ Reviewer ]
@atigyi 

[ QA ]
Peer reviewed
Spell checked 
Clicked all links

[ Release Notes ]
Updates to the User Guide to reflect the new Open Source porting process.  Also added links to our CC3200 and STM32 Nucleo examples, and to our Further Readings.

### CODE REVIEW

File: doc/user_guide.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 # Xively Client for C User Guide
-##### Copyright © 2003-2016 LogMeIn, Inc.
+##### Copyright © 2003-2017 LogMeIn, Inc.
 
 ## Hello
 
-This document is meant to be a general guide for the features and use of the Xively Client Agent written for C.  This document goes through what you should expect when you receive the Xively Client, how you should write your Client Application to use the Xively Client Agent to connect to the Xively Service, and how you can compile and link against the provided deliverables.
-This document does not provide a full source code example. However, the Xively Client build deliverable will include some examples that have been well-documented. Additionally a Doxygen API reference of our forward facing functions is provided.
-If you have any questions, comments or if you require help then please contact Xively-Client-Support <Xively-Client-Support@logmein.com> .
+This document is descibes the general features and intended use of the Xivley C Client by C applications attempting to connect to the Xively Service.  Details include the intended application flow, platform security considerations, and an API overview are contained within.
```
Comment (atigyi) [inline doc/user_guide.md]: no need for "is"

File: doc/user_guide.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+
+This document does not provide a full source code example. However, the `examples/` directory of the [Xively C Client github repository](https://github.com/xively/xively-client-c/tree/master) includes many examples for POSIX platforms, and ST and TI reference boards.
```
Comment (atigyi) [inline doc/user_guide.md]: and ST and TI --> ST and TI

File: doc/user_guide.md [hunk 3]
```diff
@@ -20,25 +20,26 @@ The Xively Client Agent is a C library that provides MQTT connectivity over TLS
...
 
-* We can operate on a single thread and our implementation runs in non-blocking mode by default.
+* It can operate on a single thread and our implementation runs in non-blocking mode by default.
 * The client does not use any CPU power when waiting for operations. Power consumption can be minimized as required.
-* A thread-safe event queue and an optional thread-pool for callbacks allow us to support more robust applications and platforms beyond the standard embedded platforms.
-* Our TLS Board Support Platform (BSP) can be used to use TLS features that are already embedded on some platforms.  Please see the Platform Security Requirements for more information.
+* A thread-safe event queue and an optional thread-pool for callbacks allows the client to support robust applications and platforms beyond the standard embedded platforms.
+* The Transport Layer Security (TLS) Board Support Package (BSP) can be used to leverage TLS features that are already on some mbedded platforms, or via software libraires like [WolfSSL](https://www.wolfssl.com) or [mbedTLS](https://tls.mbed.org/).  Please see the Platform Security Requirements for more information about TLS.
```
Comment (atigyi) [inline doc/user_guide.md]: 👍

File: doc/user_guide.md [hunk 4]
```diff
@@ -20,25 +20,26 @@ The Xively Client Agent is a C library that provides MQTT connectivity over TLS
...
 
-The Xively Client Back-Off system provided intelligent networking behavior to prevent fleets of devices from causing unintentional DDoS attacks on Xively services.
+The Xively Client Back-Off system provides intelligent networking behavior to prevent fleets of devices from causing unintentional DDoS attacks on Xively services.
 
-* Client-side prevention of devices from attempting to connect in a tight loop.
-* Client Application is informed for pending connection attempts, disconnects, and Back-Off status.
-* Behavior is configurable per platform so that we can ensure that we get the best user experience.
+* Prevents invdividual clients from attempting reconnections in a tight-loop.
+* Client Application is informed of pending connection attempts, disconnects, and Back-Off status changes.
+* Behavior is configurable per platform so that you can ensure that we get the best user experience.
```
Comment (atigyi) [inline doc/user_guide.md]: we get the best user experience. ?
we?

File: doc/user_guide.md [hunk 5]
```diff
@@ -47,54 +48,45 @@ The Xively Client was written with the understanding that the IoT Field might ch
...
 
 # TLS Implementation Requirements
 
-The Xively Client has been tested against WolfSSL's CyaSSL library v3.2.0 and mbedTLS (formerly PolarSSL https://tls.mbed.org). Xively Professional Services (XPS) can work with your Client Application team and with WolfSSL to port that version of the CyaSSL implementation to your platform as part of the standard porting process. If your platform software or hardware has a different TLS implementation already provided, then Xively Professional Services can work with your Client Application team to use your TLS version.
+The Xively Client has been tested against [WolfSSL's library v3.2.0](https://www.wolfssl.com) and [mbedTLS](https://tls.mbed.org) (formerly PolarSSL ). If your platform e has a different TLS implementation then the following are guidelines for ensuring that your TLS implementaiton meets the same standards. Please see our [Xively C Client Porting Guide](https://github.com/xively/xively-client-c/blob/master/doc/porting_guide.md) for building the Xively C Client's Abstracted TLS BSP to support other TLS implementations.
```
Comment (atigyi) [inline doc/user_guide.md]: If your platform e has... --> remove "e"

Comment (atigyi) [inline doc/user_guide.md]: just an extra space after PolarSSL and before the )

Comment (atigyi) [inline doc/user_guide.md]: remove or add words or rephrase :)
"then the following are guidelines for ensuring"

File: doc/user_guide.md [hunk 6]
```diff
@@ -103,13 +95,14 @@ Note: We've found that some versions of TLS in hardware do not have all of these
 * Server Domain Checking
     * Wildcard support for domain name checks
 
-* Online Certificate Status Protocol (OCSP): http://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol
+* [Online Certificate Status Protocol (OCSP)](http://en.wikipedia.org/wiki/Online_Certificate_Status_Protocol)
```
Comment (atigyi) [inline doc/user_guide.md]: 👍

File: doc/user_guide.md (hunk 4 above)
Comment (DellaBitta) [inline doc/user_guide.md]: Trump wrote it.

File: doc/user_guide.md [hunk 7]
```diff
@@ -123,7 +116,8 @@ Note: We've found that some versions of TLS in hardware do not have all of these
 
 # Xively Client Standard Features
 
-The Xively Client Agent for C was designed with embedded systems in mind. That is, it is a low footprint, low memory usage, communications solution that can run on a single thread with a single socket connection on multiple operating systems. But the client itself also offers a suite of subsystems that go beyond a standard communications API.
+The Xively Client Agent for C was designed with embedded systems in mind. It has a low footprint, minimal memory usage, and a scheduling system to make your application's use of MQTT as easy as possible.  It can run on a single thread with a single socket connection on multiple operating systems.  The following are details about all of these subsystems.
```
Comment (atigyi) [inline doc/user_guide.md]: "The following are details about all of these subsystems."
The following details are about ... ?

File: doc/user_guide.md [hunk 8]
```diff
@@ -496,33 +480,36 @@ if( XI_INVALID_TIMED_TASK_HANDLE != timed_task_handle )
...
-
-Provided as part of your Xively Port Deliverable.  Please check the release notes for any bug fixes, API changes, etc. to ensure that your Client Application behaves as expected.
 
 # External Resources
 
 ## MQTT
 
-* https://en.wikipedia.org/wiki/MQTT
-* http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html
+* [MQTT Wikipedia](https://en.wikipedia.org/wiki/MQTT)
```
Comment (atigyi) [inline doc/user_guide.md]: 👍

File: doc/user_guide.md [hunk 9]
```diff
@@ -210,25 +194,25 @@ Queries the current amount of memory allocated for Xively Client operations, inc
...
-Before executing the Xively Client you must coordinate with Xively Professional Services to create Xively Client connection credentials for your application. The following credentials must be provided to the Xively Client for C during its initialization, connection and subscription/publish requests:
+Before executing the Xively Client you must create Xively Client connection credentials for your application. The following credentials can be created in your account at [app.xivley.com](https://app.xively.com):
 
 - account id
 - device id
 - device secret (password)
 - topic name
 
-Xively Professional Services will either provide access to the Xively Blueprint System so that you may create credentials yourselves, or they will provide you with pre-created credentials with which you may begin your testing and evaluation.
+For more information about using the Xively cloud service please see the [Xively Developer Center](https://developer.xively.com/).
```
Comment (atigyi) [inline doc/user_guide.md]: Capital letters? Xively Cloud Service?

**Comment by atigyi** [comment]
There are also two occurrences of cyassl if you feel the power then pls change them to wolfSSL, thx.

**Pull request merged**